hazelcast/src/main/java/com/hazelcast/map/impl/MapConfigValidator.java

```diff
@@ -36,36 +38,54 @@ private MapConfigValidator() {
     }
 
     /**
-     * Throws {@link IllegalArgumentException} if the supplied {@link InMemoryFormat} is {@link InMemoryFormat#NATIVE}
+     * Checks preconditions to create a map proxy.
      *
-     * @param inMemoryFormat supplied inMemoryFormat.
+     * @param mapConfig the {@link MapConfig}
      */
-    public static void checkNotNative(InMemoryFormat inMemoryFormat) {
-        if (NATIVE == inMemoryFormat) {
-            throw new IllegalArgumentException("NATIVE storage format is supported in Hazelcast Enterprise only. "
-                    + "Make sure you have Hazelcast Enterprise JARs on your classpath!");
-        }
+    public static void checkMapConfig(MapConfig mapConfig) {
+        checkNotNative(mapConfig.getInMemoryFormat());
+
+        logIgnoredConfig(mapConfig);
     }
 
     /**
-     * Checks preconditions to create a map proxy.
+     * Checks preconditions to create a map proxy with Near Cache.
      *
-     * @param mapConfig the mapConfig
+     * @param nearCacheConfig the {@link NearCacheConfig}
+     * @param isClient        {@code true} if the config is for a Hazelcast client, {@code false} otherwise
      */
-    public static void checkMapConfig(MapConfig mapConfig) {
-        checkNotNative(mapConfig.getInMemoryFormat());
+    public static void checkNearCacheConfig(NearCacheConfig nearCacheConfig, boolean isClient) {
+        checkNotNative(nearCacheConfig.getInMemoryFormat());
 
-        logIgnoredConfig(mapConfig);
+        logIgnoredConfig(nearCacheConfig, isClient);
+    }
+
+    /**
+     * Throws {@link IllegalArgumentException} if the supplied {@link InMemoryFormat} is {@link InMemoryFormat#NATIVE}
+     *
+     * @param inMemoryFormat supplied inMemoryFormat
+     */
+    private static void checkNotNative(InMemoryFormat inMemoryFormat) {
+        if (inMemoryFormat == NATIVE && !getBuildInfo().isEnterprise()) {
```

Comment:
I had to add this check for Enterprise, since the `AbstractClientInternalCacheProxy` is also used in Hazelcast Enterprise, which of course works fine with `NATIVE`.